lib/index.d.ts

```diff
@@ -153,7 +153,8 @@ export class Component<
   AttrsT = AnyAttrs,
   AppStateT = unknown,
   AppT = unknown,
-  ContextRegistryT = unknown
+  ContextRegistryT = unknown,
+  HelpersT extends PanelHelpers = unknown
```

Comment:
In order to take advantage of this, you'll need to include a component-specific helpers type to fill in the generic for each individual component, right? Given that we're moving away from using `helpers` in general (it was mainly there for Jade/Pug ergonomics, and is mostly just extra noise in already-noisy TSX), I wonder if it's worth adding these helpers type defs everywhere. The better move may be to spend that effort moving things out of helpers into main component methods.

Reply:
OK, if it doesn't need anything except this unused generic, I'm fine with adding it, at least until `helpers` goes away. Just get the build passing and we can merge.

Reply:
can we just type it with `&`?

```
get config(): ReturnType<Component['config']> & Helpers
```

another generic on the class is pretty messy indeed.